src/wp-includes/blocks.php

```diff
@@ -1358,3 +1358,96 @@ function _wp_multiple_block_styles( $metadata ) {
 	return $metadata;
 }
 add_filter( 'block_type_metadata', '_wp_multiple_block_styles' );
+
+/**
+ * Helper function that constructs a comment query vars array from the passed
+ * block properties.
+ *
+ * It's used with the Comment Query Loop inner blocks.
+ *
+ * @since 6.0.0
+ *
+ * @param WP_Block $block Block instance.
+ *
+ * @return array Returns the comment query parameters to use with the
+ * WP_Comment_Query constructor.
+ */
+function build_comment_query_vars_from_block( $block ) {
+
+	$comment_args = array(
+		'orderby'                   => 'comment_date_gmt',
+		'order'                     => 'ASC',
+		'status'                    => 'approve',
+		'no_found_rows'             => false,
+		'update_comment_meta_cache' => false, // We lazy-load comment meta for performance.
+	);
+
+	if ( ! empty( $block->context['postId'] ) ) {
+		$comment_args['post_id'] = (int) $block->context['postId'];
+	}
+
+	if ( get_option( 'thread_comments' ) ) {
+		$comment_args['hierarchical'] = 'threaded';
+	} else {
+		$comment_args['hierarchical'] = false;
+	}
+
+	$per_page     = get_option( 'comments_per_page' );
+	$default_page = get_option( 'default_comments_page' );
+
+	if ( $per_page > 0 ) {
+		$comment_args['number'] = $per_page;
+
+		$page = (int) get_query_var( 'cpage' );
+		if ( $page ) {
+			$comment_args['paged'] = $page;
+		} elseif ( 'oldest' === $default_page ) {
+			$comment_args['paged'] = 1;
+		} elseif ( 'newest' === $default_page ) {
+			$comment_args['paged'] = (int) ( new WP_Comment_Query( $comment_args ) )->max_num_pages;
+		}
+		// Set the `cpage` query var to ensure the previous and next pagination links are correct
+		// when inheriting the Discussion Settings.
+		if ( 0 === $page && isset( $comment_args['paged'] ) && $comment_args['paged'] > 0 ) {
+			set_query_var( 'cpage', $comment_args['paged'] );
```

Comment:
Why write to a query variable here? It breaks everything else in WP. It should remain `0` on the first page, as it always had been.

Reply:
Cross-referencing https://github.com/WordPress/wordpress-develop/pull/6291, where this is now being discussed, and https://github.com/WordPress/gutenberg/pull/39227, where it was originally introduced.